Lib/fontbakery/profiles/googlefonts.py

```diff
@@ -6396,42 +6396,60 @@ def com_google_fonts_check_metadata_category_hint(family_metadata):
-        an SVG color table. However, Colr v1 is only well supported in Chrome so
-        we need to add an SVG table to these fonts.
+        an SVG color table. However, Colr v1 is currently not broadly supported
+        so we need to add an SVG table to these fonts, except in the case of
+        variable fonts, since SVG does not support OpenType Variations.
 
-        To add an SVG table, run the maximum_color tool in Nano Emoji,
+        To add an SVG table, run the maximum_color tool in nanoemoji:
         https://github.com/googlefonts/nanoemoji
     """,
-    proposal = 'https://github.com/googlefonts/fontbakery/issues/3886'
+    proposal = ['https://googlefonts.github.io/gf-guide/color.html',
+                'https://github.com/googlefonts/fontbakery/issues/3886',
+                'https://github.com/googlefonts/fontbakery/issues/3888',
+                'https://github.com/googlefonts/fontbakery/pull/3889',
+                'https://github.com/googlefonts/fontbakery/issues/4131']
 )
 def com_google_fonts_check_colorfont_tables(ttFont):
-    """Check font has the expected color font tables"""
+    """Check font has the expected color font tables."""
+    passed = True
+
     if "COLR" in ttFont:
-        colr_table = ttFont["COLR"]
-        if colr_table.version == 0 and "SVG " in ttFont:
-            yield FAIL, Message(
-                "drop-svg",
-                "Font has a COLR v0 table, which is already widely supported, "
-                "so the SVG table isn't needed."
-            )
-            return
-        elif colr_table.version == 1 and "SVG " not in ttFont:
-            yield FAIL, Message(
-                "add-svg",
-                "Font has COLRv1 but no SVG table; for CORLv1, we require "
-                "that an SVG table is present to support environments where "
-                "the former is not supported yet."
-            )
-            return
-    elif "SVG " in ttFont:
+        if ttFont["COLR"].version == 0 and "SVG " in ttFont:
+            passed = False
+            yield FAIL,\
+                  Message("drop-svg",
+                          "Font has a COLR v0 table, which is already widely supported,"
+                          " so the SVG table isn't needed.")
+
+        elif ttFont["COLR"].version == 1 and "SVG " not in ttFont:
+            passed = False
+            yield FAIL,\
+                  Message("add-svg",
+                          "Font has COLRv1 but no SVG table; for CORLv1, we require"
+                          " that an SVG table is present to support environments where"
+                          " the former is not supported yet.\n"
+                          "You can do it by using the maximum_color tool provided by"
+                          " the nanoemoji project:\n"
+                          "https://github.com/googlefonts/nanoemoji")
+
+    if "SVG " in ttFont:
+        if is_variable_font(ttFont):
```

Comment:
I believe you simply check if fvar is in the ttFont to determine if a font is "variable", is that right?
While that's a good euristic in general, for this particular case I think it might be too restrictive. It's true that a VF must have at least the fvar table, but a font can be variable in many ways. It is conceivable that a font might contain _no_ glyph outline variations (gvar) while having variations in the glyph metrics (HVAR) or in font metrics (MVAR) or in GPOS, etc. In this case a (static) SVG table (which only provides the glyph shapes) would not be incompatible the variable font.
It's only when there are some glyph outline variations that the SVG table becomes incompatible with a variable font.
Perhaps you could check if the font has gvar and if that gvar is not empty.
Though even the latter check might be too restrictive; it is possible that the gvar contain variations for glyphs that are not also specified in the SVG table, in the sense that the two tables do not overlap, or in other words, the variable glyphs are not also SVG glyphs and viceversa. In this case, SVG table would not be incompatible with VF with non-empty gvar table...

Reply:
I understand that what you describe is technically possible but do we want to check for the corner-cases before actually getting samples of them in real font projects?

And if there are such cases, then I'd presume that the font designer/engineer will be already aware of what they're doing to carefully craft this compatible combination of SVG and variations. Worst case: they read the FAIL message, disagree and hopefully let us know ;-)